### ISSUE TITLE
This is a bug on xgboost.

### ISSUE THREAD

**Issue by Columbine21** [TeamHG-Memex/eli5#392]
**Description: XGBClassifier explainer failed when n_estimators=1.(extreme situation)**
E.g.
model_t = XGBClassifier(random_state=1111, max_depth=4, n_estimators=1)

show_prediction(model, test_input[tougue_correct_q[0]]) will failed to run.

**The error message is as follows:**
<img width="896" alt="截屏2020-10-21 上午9 18 32" src="https://user-images.githubusercontent.com/46373350/96661334-65d1fe00-137e-11eb-952f-6c8d989b6315.png">

**Comment by pjgao** [comment]
same errors, the problem lies in :
```python
def _prediction_feature_weights(booster, dmatrix, n_targets,
                                feature_names, xgb_feature_names):
    """ For each target, return score and numpy array with feature weights
    on this prediction, following an idea from
    http://blog.datadive.net/interpreting-random-forests/
    """
    # XGBClassifier does not have pred_leaf argument, so use booster
    leaf_ids, = booster.predict(dmatrix, pred_leaf=True)
```
`booster.predict(dmatrix, pred_leaf=True)` will return a 1-d array when only one tree in `xgboost`.  
this may be modified to:
```python
def _prediction_feature_weights(booster, dmatrix, n_targets,
                                feature_names, xgb_feature_names):
    """ For each target, return score and numpy array with feature weights
    on this prediction, following an idea from
    http://blog.datadive.net/interpreting-random-forests/
    """
    # XGBClassifier does not have pred_leaf argument, so use booster
    leaf_ids, = booster.predict(dmatrix, pred_leaf=True).reshape(1,-1)
```

**Comment by pjgao** [comment]
Now the question is: Is `dmatrix` only contains one sample ?
